.github/workflows/build.yml

```diff
@@ -0,0 +1,321 @@
+name: build
+
+on:
+  push:
+    branches:
+    - master
+    tags:
+    - v*
+  pull_request:
+    branches:
+    - master
+  schedule:
+    - cron:  '23 1 * * 0'
+  release:
+    types: [published]
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group: build-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
This concurrency group will automatically cancel previous runs if a run on the _same branch_ is triggered.